src/lib/blockchain/insuranceOracle.ts

```
@@ -10,11 +10,11 @@ const supabase = createClient(
  * Fetches real-time weather data from NOAA API (mocked for this implementation).
  * In production, this would be a Chainlink External Adapter or dedicated Oracle node.
  */
-export async function fetchNOAAWeatherData(lat: number, lon: number): Promise<number> {
+export async function fetchNOAAWeatherProbability(lat: number, lon: number): Promise<number> {
     // Mock NOAA API response
     // In production: const response = await fetch(`https://api.weather.gov/points/${lat},${lon}`);
-    const mockPrecipitationInches = 1.25; // Simulating a hurricane/heavy rain event
-    return mockPrecipitationInches;
+    const mockProbability = 75; // Simulating 75% TWAP rain probability
```

ESLint

[error] Replace `····` with `··`

(prettier/prettier)